### ISSUE TITLE
Bug/issue using a `pnpm-workspace.yaml`

### ISSUE THREAD

**Issue by KATT** [webpro/knip#284]
Hey, I used Knip and everything worked fine.

Then I added a `pnpm-workspace.yaml` in https://github.com/KATT/tupleson/pull/34 _(aka this is where it started failing)_ which led me to delete Knip in https://github.com/KATT/tupleson/pull/35 & now I'm trying to re-add it in https://github.com/KATT/tupleson/pull/40.

It's a pnpm workspace but only has a root `package.json` and an example folder.

It starts reporting unused files and stuff: https://github.com/KATT/tupleson/pull/40/files#r1348777005

I feel something is wrong with my setup. Any assistance possible? 🙏

**Comment by webpro** [comment]
I can dig into it later if necessary, but I saw a space in `"entry": ["src/{server,client, index}.ts!"]` which could be troublesome.

**Comment by KATT** [comment]
I deleted the space now but it's unrelated. That whole workspaces-addition was me trying to fix it - it doesn't work when ignoring the whole examples folder either and picks up a bunch of false-negatives from the root project. Examples:

```graphql
Unused files (1)
tsup.config.ts # <-- used for build, didn't complain prior to PR #34
Unused devDependencies (3)
@js-temporal/polyfill  package.json # <-- used in a test.ts
@types/eslint          package.json # <--- lint is in package.json
decimal.js             package.json # <-- used in a test.ts
[...]

```

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.31.0. See [Release 2.31.0](https://github.com/webpro/knip/releases/tag/2.31.0) for release notes._

**Comment by webpro** [comment]
Guess it was a bit rare to explicitly include the root workspace in `pnpm-workspace.yaml`, but should be OK now.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.32.0. See [Release 2.32.0](https://github.com/webpro/knip/releases/tag/2.32.0) for release notes._
